app/models/shipit/pull_request.rb

```diff
@@ -163,6 +163,8 @@ def merge!
         commit_message: 'Merged by Shipit',
         merge_method: 'merge',
       )
+      stack.undeployed_commits_count += 1
```

Comment:
> Also I believe this should be done as part of the `complete!` state transition
Can you share an example of that? I see the `event :complete do` block but not sure if/how I'd access instance information inside it.